custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -233,6 +233,26 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
                 }
             };
 
+            //Creating a promise, which resolves only after map is rendered (wait until canvas is rendered in dom)
+            // Reference: https://stackoverflow.com/a/47776379/12839195 (using timeout instead of rafAsync)
+            vm.checkElement = function (selector) {
+                // if it is not a page where map is displayed, resolve the promise immediately,
+                // else wait till canvas (svg in dom) is created
+                if (!navigationService.isMapDisplayed($location.path()) && window.Promise) {
+                    return window.Promise.resolve(true);
+                }
+                if (document.querySelector(selector) === null) {
+                    return $timeout().then(function () {
+                        return vm.checkElement(selector);
+                    });
+                } else {
+                    return window.Promise.resolve(true);
+                }
+            };
+
+            vm.mapLoadingPromise = vm.checkElement('svg').then(function () {
+            });
```

Comment:
can you explain why the `then` is empty here and what that achieves? Like why can't this just be `vm.checkElement('svg')`

Reply:
It throws error if promise is not handled.